packages/cli/src/report/report-disk-writer.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import * as filenamify from 'filenamify-url';
+import * as fs from 'fs';
+import { injectable } from 'inversify';
+import { isEmpty } from 'lodash';
+import { ReportFormats } from './report-formats';
+
+@injectable()
+export class ReportDiskWriter {
+    constructor(private readonly fileSystemObj: typeof fs = fs) {}
+
+    public writeToDirectory(directory: string, fileName: string, format: ReportFormats, content: string): void {
+        if (isEmpty(directory)) {
+            // tslint:disable-next-line: no-parameter-reassignment
+            directory = '.';
+        }
+
+        const reportFileName = `${directory}/${filenamify(fileName, { replacement: '_' })}.${format}`;
```

Comment:
we may not be able to use url value as file name since you may get path too long exceptions. (but this doesn't have to be solved in this pr)

Reply:
okies, we can look at this in another PR